src/Escaper/Escaper.c

```diff
@@ -0,0 +1,127 @@
+/**@file
+ * This file is part of the TASTE Runtime Common.
+ *
+ * @copyright 2021 N7 Space Sp. z o.o.
+ *
+ * TASTE Runtime Common was developed under a programme of,
+ * and funded by, the European Space Agency (the "ESA").
+ *
+ * Licensed under the ESA Public License (ESA-PL) Permissive,
+ * Version 2.3 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://essr.esa.int/license/list
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "Escaper.h"
+#include "Broker.h"
+
+#include <assert.h>
+
+void
+Escaper_start_decoder(Escaper* const self)
+{
+    self->m_parse_state = STATE_WAIT;
+}
+
+void
+Escaper_decode_packet(Escaper* const self, uint8_t* buffer, const size_t length, Receive_packet receivePacket)
+{
+    for(size_t i = 0; i < length; ++i) {
+        switch(self->m_parse_state) {
+            case STATE_WAIT:
+                if(buffer[i] == START_BYTE) {
+                    self->m_parse_state = STATE_DATA_BYTE;
+                }
+                break;
+            case STATE_DATA_BYTE:
+                if(buffer[i] == STOP_BYTE) {
+                    receivePacket(self->m_recv_packet_buffer, self->m_recv_packet_buffer_index);
+                    self->m_recv_packet_buffer_index = 0;
+                    self->m_parse_state = STATE_WAIT;
+                } else if(buffer[i] == ESCAPE_BYTE) {
+                    self->m_parse_state = STATE_ESCAPE_BYTE;
+                } else if(buffer[i] == START_BYTE) {
+                    self->m_recv_packet_buffer_index = 0;
+                    self->m_parse_state = STATE_DATA_BYTE;
+                } else {
+                    self->m_recv_packet_buffer[self->m_recv_packet_buffer_index] = buffer[i];
+                    ++self->m_recv_packet_buffer_index;
+                }
+                break;
+            case STATE_ESCAPE_BYTE:
+                self->m_recv_packet_buffer[self->m_recv_packet_buffer_index] = buffer[i];
+                ++self->m_recv_packet_buffer_index;
+                self->m_parse_state = STATE_DATA_BYTE;
+                break;
+            default:
+                assert("Unknown parser state");
+                break;
+        }
+        assert(self->m_recv_packet_buffer_index < sizeof(self->m_recv_packet_buffer));
```

Comment:
assert?

Reply:
Yes, @Lurkerpas  recommended using assert to ensure that size of m_recv_packet_buffer table is not exceeded

Reply:
Ok, I can accept this.